src/HealthGPS/dummy_model.cpp

```diff
@@ -27,26 +28,32 @@ void DummyModel::update_risk_factors(RuntimeContext &context) {
             continue;
         }
 
-        set_risk_factors(entity, context.scenario().type());
+        set_risk_factors(entity, context);
     }
 }
 
-void DummyModel::set_risk_factors(Person &person, ScenarioType scenario) const {
-    for (auto i = 0u; i < names_.size(); ++i) {
+void DummyModel::set_risk_factors(Person &person, RuntimeContext &context) const {
```

Comment:
I would pass `person`, `scenario` and `time_elapsed`. That;s easier to work with (to understand and test) than passing the whole context. It is also more efficent as you only need to calculate such a time once instead of once per risk factor.

Reply:
Done!